### PR TITLE
refactor(dicehub):  refactor publish-item to dop provider

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
refactor publish-item to dop provider

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?id=322452&iterationID=1354&pId=322436&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     refactor publish-item to dop provider         |
| 🇨🇳 中文    |       publish-item 模块化迁移到dop       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: cmd/erda-server/main.go [hunk 1]
```diff
@@ -124,6 +123,9 @@ import (
 	_ "github.com/erda-project/erda/internal/apps/dop/providers/pipelinetemplate"
 	_ "github.com/erda-project/erda/internal/apps/dop/providers/project/home"
 	_ "github.com/erda-project/erda/internal/apps/dop/providers/projectpipeline"
+
+	// dop
```
Comment (sfwn) [inline cmd/erda-server/main.go:127]: incorrect movement

File: internal/apps/dop/provider.go [hunk 2]
```diff
@@ -96,10 +97,11 @@ type provider struct {
...
 	DevFlowRule           devflowrule.Interface
-	TokenService          tokenpb.TokenServiceServer     `autowired:"erda.core.token.TokenService"`
-	ClusterSvc            clusterpb.ClusterServiceServer `autowired:"erda.core.clustermanager.cluster.ClusterService"`
-	DevFlowSvc            *flow.Service                  `autowired:"erda.apps.devflow.flow.FlowService"`
-	IssueCoreSvc          *core.IssueService             `autowired:"erda.dop.issue.core.IssueCoreService"`
+	TokenService          tokenpb.TokenServiceServer      `autowired:"erda.core.token.TokenService"`
+	ClusterSvc            clusterpb.ClusterServiceServer  `autowired:"erda.core.clustermanager.cluster.ClusterService"`
+	DevFlowSvc            *flow.Service                   `autowired:"erda.apps.devflow.flow.FlowService"`
+	IssueCoreSvc          *core.IssueService              `autowired:"erda.dop.issue.core.IssueCoreService"`
+	PublishItemSvc        *publishitem.PublishItemService `autowired:"erda.dop.publishitem.PublishItemService"`
```
Comment (sfwn) [inline internal/apps/dop/provider.go:104]: what's the difference between inject `*publishitem.PublishItemService` and `pb.PublishItemServer`?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: cmd/erda-server/main.go (hunk 1 above)
Comment (chengjoey) [reply cmd/erda-server/main.go:127]: adjusted

File: internal/apps/dop/provider.go (hunk 2 above)
Comment (chengjoey) [reply internal/apps/dop/provider.go:104]: adjusted, remove no usage publishitemsvc

**Review by sfwn — APPROVED** [review]

**Pull request merged**